Runtime/Scripts/RGBotConfigs/RGButtonState.cs

```diff
@@ -9,9 +9,9 @@ public class RGButtonState : RGState
 
         public RGButtonState()
         {
-            if (string.IsNullOrEmpty(this.objectType))
+            if (string.IsNullOrEmpty(rgEntity.objectType))
```

Comment:
I think if you try out the menu and ability bot for BossRoom with these changes you will find it to be broken.

This change would fundamentally alter how we had buttons modeled.

There is only 1 Agent for buttons and it is attached to the RGOverlayCanvas.  Any RGButtonState on any canvas is mapped to that 1 Agent.  Thus all UI button input is managed through a single Agent which basically acts as the `Human User` that would click said buttons.

I would expect that RGAgent on the RGOverlayCanvas would be changed to an RGEntity for this set of changes, but I would NOT expect every button to require an RGEntity on it as that would change the Entity/Agent simulating a single human user model.